### DISCUSSION TITLE
Gauge plugin crashes on Visual Studio Code

### DISCUSSION THREAD

**Discussion by esikder** [getgauge/gauge#2139] [General]
Hi,
Is anyone facing a problem with the gauge plugin on VS Code?
I am getting:
[Info  - 2:55:49 PM] Connection to server got closed. Server will restart.
[Info  - 2:55:49 PM] Connection to server got closed. Server will restart.
[Info  - 2:55:49 PM] Connection to server got closed. Server will restart.
[Info  - 2:55:49 PM] Connection to server got closed. Server will restart.
[Error - 2:55:49 PM] Connection to server got closed. Server will not be restarted.
![image](https://user-images.githubusercontent.com/21352658/138806713-475ae2b7-a5d2-4579-8c77-89c4066b8440.png)

I am using MAC OS big sur version 11.6 and Visual Studio Code Version: 1.61.2 (Universal), gauge plugin is crashing at the start and I cant use the plugin. It was working fine just started failing around 3 weeks ago.
Is there any new change?

version:
Gauge version: 1.1.8
Commit Hash: b1501f4

Plugins
csharp (0.10.6)
dotnet (0.3.1)
flash (0.0.2)
html-report (4.0.12)
screenshot (0.0.1)
spectacle (0.1.4)
xml-report (0.2.3)

**Comment by sriv** [comment]
If you haven't changed gauge/plugins then chances are something changed in your environment/language tooling.

Couple of things to try:
- Can you try updating gauge and the plugins please? Gauge 1.4.2 is the latest release.
- inspect the logs (gauge.log, lsp.log) under the logs directory. Do you see any errors?
- try running `gauge daemon --lsp` from the project root in terminal to see if there are any errors.

**Reply by esikder** [reply]
Thanks Sriv for the quick response. It's not only my machine only, it is for everyone in my company.
I have upgraded the gauge and the following is the version now, plugin is the latest and have latest VS Code
Gauge version: 1.4.2
Commit Hash: f080fbc

Plugins
-------
csharp (0.10.6)
dotnet (0.4.1)
flash (0.0.2)
html-report (4.0.12)
screenshot (0.1.0)
spectacle (0.1.4)
xml-report (0.2.3)


following is the return from gauge daemon --lsp 

{"method":"window/showMessage","params":{"type":1,"message":"Gauge could not initialize. Install 'ms-dotnettools.csharp' extension for code insights. For more information see[Problems](command:workbench.actions.view.problems), check logs.[Troubleshooting](https://docs.gauge.org/troubleshooting.html?language=javascript\u0026ide=vscode#gauge-could-not-initialize-for-more-information-see-problems)"},"jsonrpc":"2.0"}


and following is the part from lsp.log
notif #0: initialized: runtime error: invalid memory address or nil pointer dereference
goroutine 65 [running]:
runtime/debug.Stack(0xc0002ff780, 0x173c140, 0x1d190b0)
	/opt/hostedtoolcache/go/1.16.9/x64/src/runtime/debug/stack.go:24 +0x9f
github.com/getgauge/gauge/api/lang.recoverPanic(0xc0000da050)
	/home/runner/work/gauge/gauge/api/lang/server.go:308 +0x5b
panic(0x173c140, 0x1d190b0)
	/opt/hostedtoolcache/go/1.16.9/x64/src/runtime/panic.go:965 +0x1b9
github.com/getgauge/gauge/runner.(*GrpcRunner).ExecuteMessageWithTimeout(...)
	/home/runner/work/gauge/gauge/runner/grpcRunner.go:195
github.com/getgauge/gauge/api/lang.globPatternRequest(0x4, 0x4, 0x16dff60)
	/home/runner/work/gauge/gauge/api/lang/runner.go:87 +0x79
github.com/getgauge/gauge/api/lang.registerRunnerCapabilities(0x1913180, 0xc000528000, 0x1915e70, 0xc0000b0008, 0x0, 0x0)
	/home/runner/work/gauge/gauge/api/lang/capabilities.go:110 +0x4b
github.com/getgauge/gauge/api/lang.(*LangHandler).Handle(0x1d63538, 0x1915e70, 0xc0000b0008, 0x1913180, 0xc000528000, 0xc0000da050, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/gauge/gauge/api/lang/server.go:76 +0x645
github.com/getgauge/gauge/api/lang.(*LangHandler).handle(0x1d63538, 0x1915e70, 0xc0000b0008, 0xc000528000, 0xc0000da050, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/gauge/gauge/api/lang/server.go:60 +0xa5
github.com/sourcegraph/jsonrpc2.(*HandlerWithErrorConfigurer).Handle(0xc000510050, 0x1915e70, 0xc0000b0008, 0xc000528000, 0xc0000da050)
	/home/runner/go/pkg/mod/github.com/sourcegraph/jsonrpc2@v0.0.0-20200429184054-15c2290dcb37/handler_with_error.go:21 +0x72
created by github.com/getgauge/gauge/api/lang.lspHandler.Handle
	/home/runner/work/gauge/gauge/api/lang/server.go:55 +0x6b
26-10-2021 16:36:04.485 [Gauge] [INFO] Initializing concepts cache with 0 concepts
26-10-2021 16:36:04.486 [Gauge] [INFO] Initializing specs cache with 0 specs
26-10-2021 16:36:04.486 [Gauge] [INFO] Initializing steps cache with 0 steps
26-10-2021 16:36:04.487 [Gauge] [INFO] Starting language runner
26-10-2021 16:36:04.488 [Gauge] [DEBUG] Watching directory: /users/erfanasikder/rokt-end-to-end-testing/specs
26-10-2021 16:36:04.488 [Gauge] [DEBUG] Found 0 files
26-10-2021 16:36:05.182 [dotnet] [INFO] Microsoft (R) Build Engine version 16.9.0+57a23d249 for .NET
26-10-2021 16:36:05.182 [dotnet] [INFO] Copyright (C) Microsoft Corporation. All rights reserved.
26-10-2021 16:36:07.170 [dotnet] [INFO] protobufs : warning : directory does not exist. [/Users/erfanasikder/rokt-end-to-end-testing/EndToEndTest.csproj]
26-10-2021 16:36:07.172 [dotnet] [INFO] Could not make proto path relative : error : protobufs/identity-api/Identity.proto: No such file or directory [/Users/erfanasikder/rokt-end-to-end-testing/EndToEndTest.csproj]
26-10-2021 16:36:07.277 [dotnet] [CRITICAL] Unable to build Project in /users/erfanasikder/rokt-end-to-end-testing
dotnet Project build failed.
Ran 'dotnet publish --runtime=osx-x64 --configuration=release --output="/users/erfanasikder/rokt-end-to-end-testing/gauge_bin"  "EndToEndTest.csproj" --verbosity=quiet'
   at Gauge.Dotnet.GaugeProjectBuilder.BuildTargetGaugeProject() in /home/runner/work/gauge-dotnet/gauge-dotnet/src/GaugeProjectBuilder.cs:line 46
   at Gauge.Dotnet.StartCommand.TryBuild() in /home/runner/work/gauge-dotnet/gauge-dotnet/src/StartCommand.cs:line 79
26-10-2021 16:36:07.281 [Gauge] [ERROR] Error occurred while waiting for runner process to finish.
Error : exit status 1
26-10-2021 16:36:07.281 [Gauge] [DEBUG] Gauge could not initialize. Install 'ms-dotnettools.csharp' extension for code insights. For more information see[Problems](command:workbench.actions.view.problems), check logs.[Troubleshooting](https://docs.gauge.org/troubleshooting.html?language=javascript&ide=vscode#gauge-could-not-initialize-for-more-information-see-problems)
Some of the gauge lsp feature will not work as expected.
26-10-2021 16:36:07.281 [Gauge] [INFO] LangServer: reading on stdin, writing on stdout

On UI following is the error:
![image](https://user-images.githubusercontent.com/21352658/138819406-3ffca3de-cc8f-4ea2-a0d8-c5a3fb2ec5a4.png)

**Comment by esikder** [comment]
Ok I think the issue is resolved for me after clonning the repo again and updating the submodules.
I am yet not sure what caused this error but currently, I can work with the plugin.
Would appreciate if you could share any information based on the log,regarding the cause of this issue.

**Comment by zabil** [comment]
```
26-10-2021 16:36:07.170 [dotnet] [INFO] protobufs : warning : directory does not exist. [/Users/erfanasikder/rokt-end-to-end-testing/EndToEndTest.csproj]
26-10-2021 16:36:07.172 [dotnet] [INFO] Could not make proto path relative : error : protobufs/identity-api/Identity.proto: No such file or directory [/Users/erfanasikder/rokt-end-to-end-testing/EndToEndTest.csproj]
```

This looks like an error where there were some files missing in the project. This cause gauge's dotnet parser to fail. The best way to solve this is look at the looks for any project configuration errors.

**Reply by esikder** [reply]
Thanks, Zabil for looking into this.
